### DISCUSSION TITLE
TimerInterrupts tutorial

### DISCUSSION THREAD

**Discussion by tgibbs99** [khoih-prog/ESP32TimerInterrupt#18] [Q&A]
I am looking for (asking for) a tutorial on Esp32timerinterrupt Library. The examples do not appear to work as expected. I would like to see an example of a single timer, that triggers only on duration, and not interval. Then an example that triggers on both interval and duration, then an example of two timers triggering only on duration. 

I would like to see some sort of flow chart showing where the program goes once the interrupt is triggered, and where it returns once the interrupt completes. 

I have been working with the TimerInterruptTest example. When I attempt to simplify the program by commenting all the lines referring to interval triggers, and leave just the duration triggers, the program compiles, but gives no output on the serial monitor. 

Thanks.

Tgibbs99

**Comment by m66rob** [comment]
> Estoy buscando (pidiendo) un tutorial sobre Esp32timerinterrupt Library. Los ejemplos no parecen funcionar como se esperaba. Me gustaría ver un ejemplo de un solo temporizador, que se active solo en la duración, y no en el intervalo. Luego, un ejemplo que se activa tanto en el intervalo como en la duración, luego un ejemplo de dos temporizadores que se activan solo en la duración.
> 
> Me gustaría ver algún tipo de diagrama de flujo que muestre a dónde va el programa una vez que se activa la interrupción y dónde regresa una vez que se completa la interrupción.
> 
> He estado trabajando con el ejemplo timerInterruptTest. Cuando intento simplificar el programa comentando todas las líneas que se refieren a los desencadenadores de intervalo, y dejo solo los desencadenadores de duración, el programa compila, pero no da salida en el monitor serie.
> 
> Gracias.
> 
> Tgibbs99

Hello, did you manage to use the examples, I also tried to use them but I could not use the tachometer
